pkg/telemetryquerylanguage/tql/boolean_value.go

```diff
@@ -74,13 +73,37 @@ func newComparisonEvaluator(comparison *Comparison, functions map[string]interfa
 		return func(ctx TransformContext) bool {
 			a := left.Get(ctx)
 			b := right.Get(ctx)
-			return a == b
+			return compare(a, b, EQ)
```

Comment:
It could, but it would require converting the lexer from the "MustSimple" form to a different style, which is possible, but I'd rather do it as a separate step. I did, however, convert this function to use a table lookup rather than a switch statement, which makes the code more compact.

Reply:
I was hoping we could add a new type and custom capture method like `IsNil`, `Boolean`, and `EnumSymbol`.  Would that be doable?  That would remove any need to a string to struct lookup.

Reply:
If not, I like your map solution.